drawBot/drawBotDrawingTools.py

```diff
@@ -1239,7 +1321,7 @@ def lineCap(self, value):
         self._requiresNewFirstPage = True
         self._addInstruction("lineCap", value)
 
-    def lineDash(self, *value, offset=0):
+    def lineDash(self, *value: list[float], offset=0):
```

Comment:
You are right, but looking at the code underneath I am bit confused by these lines:
https://github.com/typemytype/drawbot/blob/f115a0ee051328efaf8bbe17e40f9f08dc3f81d1/drawBot/drawBotDrawingTools.py#L1282-L1283

Was it an older interface? If yes, should we add a warning? cc @typemytype

Reply:
Oh hmm, it seems to accept both `lineDash([a, b, c])` as well as `lineDash(a, b, c)`. If we want to keep this, we should annotate as:

```python
    def lineDash(self, *value: float | list[float], offset: float = 0):
```

Reply:
We don't verify that len(value) == 1 in the case of `lineDash([a, b, c])`. That case would warrant a TypeError.

We may want to deprecate the list spelling, but let's consider that a separate issue.

Reply:
This section changed quite a bit since your comment. We update the code and the method signature. It looks like this now:
```python
def lineDash(self, value: float | None, *values: float, offset: float = 0):
    combinedDashes: list[float | None] = [value]
    if isinstance(value, float | int):
        combinedDashes.extend(values)
    self._requiresNewFirstPage = True
    self._addInstruction("lineDash", combinedDashes, offset)
```